src/common/inc/CaPTkUtils.h

```diff
@@ -111,7 +112,7 @@ struct NonNativeApp
 */
 inline int guessImageType(const std::string &fileName)
 {
-	std::string basename = cbica::getFilenameBase(fileName);
+	std::string basename = cbica::getFilenameBase(fileName, false); // don't check the file for existence
```

Comment:
This is incorrect, as we should be checking for the presence of file for this function.

Reply:
I don't see why this is relevant for the guessImageType function, though. Intuitively, I expect to be able to pass a filename to guessImageType and have it do exactly that: examine the name and return an image type. Nor does it have documentation stating that you should only pass a full path to it.
The reason this was crashing before is that guessImageType was being called (from the "LoadQualifiedSubjectsFromGivenDirectory" family of functions in fMainWindow) with _just_ a filename, like AAAA_PH.nii.gz or something. Then this was passed to getFilenameBase which by default passed it to fileExists. But fileExists can't work on just a filename, it needs the path to find it. 

Are there other parts of the code relying on guessImageType to do the existence checking? Shouldn't those parts be checking existence separately? It could be easy to just change the calling code in fMainWindow to pass the full filepath to guessImageType. But I really don't think we should be relying upon side effects of it unless it's more explicit. 

Perhaps I could just overload guessImageType with a bool that determines whether the file is checked, and have it default to true (as is currently the case)? This way, no existing code would break, but we can still call guessImageType on path-less filenames without it exiting.

Reply:
This function is to try and guess the modality for the supplied image file, therefore, it is necessary for the file to be present. 

> The reason this was crashing before is that guessImageType was being called (from the "LoadQualifiedSubjectsFromGivenDirectory" family of functions in fMainWindow) with just a filename, like AAAA_PH.nii.gz or something

I would still say that this function should be called with the full-qualified file path. 

> Perhaps I could just overload guessImageType with a bool

This is a good intermediate solution.

Reply:
Done.